src/reactor/fns.js

```diff
@@ -223,65 +226,79 @@ export function getOption(reactorState, option) {
 
 /**
  * Use cases
- * removeObserver(observerState, [])
- * removeObserver(observerState, [], handler)
- * removeObserver(observerState, ['keyPath'])
- * removeObserver(observerState, ['keyPath'], handler)
- * removeObserver(observerState, getter)
- * removeObserver(observerState, getter, handler)
+ * removeObserver(observerState, reactorState, [])
+ * removeObserver(observerState, reactorState, [], handler)
+ * removeObserver(observerState, reactorState, ['keyPath'])
+ * removeObserver(observerState, reactorState, ['keyPath'], handler)
+ * removeObserver(observerState, reactorState, getter)
+ * removeObserver(observerState, reactorState, getter, handler)
  * @param {ObserverState} observerState
+ * @param {ReactorState} reactorState
  * @param {KeyPath|Getter} getter
  * @param {Function} handler
- * @return {ObserverState}
+ * @return {Array}
  */
-export function removeObserver(observerState, getter, handler) {
+export function removeObserver(observerState, reactorState, getter, handler) {
   const entriesToRemove = observerState.get('observersMap').filter(entry => {
     // use the getterKey in the case of a keyPath is transformed to a getter in addObserver
-    let entryGetter = entry.get('getterKey')
+    const entryGetter = entry.get('getterKey')
+    const entryGetterLiteral = convertToGetterLiteral(entryGetter)
     let handlersMatch = (!handler || entry.get('handler') === handler)
     if (!handlersMatch) {
       return false
     }
     // check for a by-value equality of keypaths
-    if (isKeyPath(getter) && isKeyPath(entryGetter)) {
-      return isEqual(getter, entryGetter)
+    if (isKeyPath(getter) && isKeyPath(entryGetterLiteral)) {
+      return isEqual(getter, entryGetterLiteral)
     }
     // we are comparing two getters do it by reference
     return (getter === entryGetter)
   })
 
-  return observerState.withMutations(map => {
-    entriesToRemove.forEach(entry => removeObserverByEntry(map, entry))
+  // Update both observer and reactor state
+  observerState = observerState.withMutations(oState => {
+    reactorState = reactorState.withMutations(rState => {
+      entriesToRemove.forEach(entry => { removeObserverByEntry(oState, rState, entry) })
```

Comment:
Hmm. I guess we'd have to keep track of the # of observers associated with a given getter. Will look into it.